components/bookmark/index.jsx

```diff
@@ -38,26 +39,31 @@ const useStyles = makeStyles((theme) => createStyles(styles(theme)));
 
 export const toggleBookmarkHandler = ({
```

Comment:
In general I would prefer if we stop exporting functionality that should be internal like this. I understand that it's easier to test functionality this way, but we should be able to test the input and output of the component without having to destructure the inner working like this.

This also creates a hellish parameter to maintain, and a lot of unnecessary wiring.

Component files should only export one function, the component itself.

Reply:
I disagree on both counts. I don't see a problem with exporting for tests, and I don't agree that component file should only ever export a single function.

Reply:
Disagree.

Reply:
I think there are ways we can make the code more logically modular than these functions, e.g. moving domain/model logic into helpers, like we've done in `src/solidClientHelpers` (I think we could probably improve these concepts even more).

I'm not opposed to these functions; they have their place, and do them if it's the best way to get something done. But I prefer to avoid them, in preference of other patterns.

Reply:
I only agree if we're re-using these functions; otherwise, why is moving the exported function from being exported from File A, to being exported from File B, better than exporting the functions from file A?

Reply:
They must be designed with reusability in mind, of course. No point in modules for modules sake.

If there really is no other ways that makes sense than to bundle some logic that allows you to test the code easier, sure, export them from the file itself.

Reply:
> They must be designed with reusability in mind, of course. No point in modules for modules sake.

Reusability isn't the only reason to create small modules; it also makes the code easy to read when it's small.

> If there really is no other ways that makes sense than to bundle some logic that allows you to test the code easier, sure, export them from the file itself.

I'm still curious about why it's worse to export small functions by default. So far it only seems to have upsides, and no downsides.

Reply:
I don't know if a function that takes an object with ten properties can be called a small function?